syntax = "proto3";

package transactionapi;

service TransactionApi {
  rpc CreateTransaction (Transaction) returns (TransactionReply) {}
  rpc GetTransaction (TransactionRequest) returns (Transaction) {}
}

message Transaction {
  string name = 1;
  uint32 id = 2;
  uint32 from = 3;
  uint32 to = 4;
  double amount = 5;
}

message TransactionReply {
  bool success = 1;
}

message TransactionRequest {
  uint32 id = 1;
}
